pkg/maps/ctmap/gc/gc.go

```diff
@@ -301,26 +301,21 @@ func (gc *GC) createGCFilter(initialScan bool, restoredEndpoints []*endpoint.End
 			}
 		}
 
-		// Once the host firewall is enabled, we will start tracking (and
-		// potentially enforcing policies) on all connections to and from the
-		// host IP addresses. Thus, we also need to avoid GCing the host IPs.
-		if option.Config.EnableHostFirewall {
-			addrs, err := nodeAddressing.IPv4().LocalAddresses()
-			if err != nil {
-				gc.logger.WithError(err).Warning("Unable to list local IPv4 addresses")
-			}
-			addrsV6, err := nodeAddressing.IPv6().LocalAddresses()
-			if err != nil {
-				gc.logger.WithError(err).Warning("Unable to list local IPv6 addresses")
-			}
-			addrs = append(addrs, addrsV6...)
+		addrs, err := nodeAddressing.IPv4().LocalAddresses()
+		if err != nil {
+			gc.logger.WithError(err).Warning("Unable to list local IPv4 addresses")
+		}
+		addrsV6, err := nodeAddressing.IPv6().LocalAddresses()
+		if err != nil {
+			gc.logger.WithError(err).Warning("Unable to list local IPv6 addresses")
+		}
+		addrs = append(addrs, addrsV6...)
```

Comment:
Shouldn't this also be guarded by the `MasqueradingEnabled` flag?

Reply:
bpf snat is not only gated by bpf masquerading flags. They are also enabled when traffic is using node ip to avoid conflict with other existing snat sessions like nodeport.

Reply:
Sure! The main point being though, BPF SNAT is done in some specific cases, so we should try to cover those cases in case of blanket removing host IPs from the GC filter. I tracked the code that tracks host traffic for the purposes of snat here - https://github.com/cilium/cilium/blob/main/bpf/lib/nodeport.h#L3104. At the very least, it seems like we should guard the GC filter code with `ENABLE_NODEPORT` flag in addition to host firewall flag.

Reply:
We can definitely carve out a list of flags to use here. But in general I'm wondering why we need to remove contrack entries corresponds to node IPs during anetd bootstrap in any case?